x-pack/plugins/cases/server/services/user_actions/operations/create.ts

```diff
@@ -52,30 +54,25 @@ export class UserActionPersister {
     this.auditLogger = new UserActionAuditLogger(this.context.auditLogger);
   }
 
-  public async bulkCreateUpdateCase({
-    originalCases,
-    updatedCases,
-    user,
-    refresh,
-  }: BulkCreateBulkUpdateCaseUserActions): Promise<void> {
-    const builtUserActions = updatedCases.reduce<UserActionEvent[]>((acc, updatedCase) => {
-      const originalCase = originalCases.find(({ id }) => id === updatedCase.id);
+  public buildUserActions({ updatedCases, user }: BuildUserActionsDictParams): UserActionsDict {
```

Comment:
An `UserActionsDict` is a dictionary where the key is the `caseId` and the value is an array of all user actions to be created.

This will help us during the validation where we can check for **each** case independently if they will break the limit.